src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts

```diff
@@ -0,0 +1,417 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { loadWorker } from '../../libs/voicefocus/loader';
+import Logger from '../logger/Logger';
+import CanvasVideoFrameBuffer from '../videoframeprocessor/CanvasVideoFrameBuffer';
+import VideoFrameBuffer from '../videoframeprocessor/VideoFrameBuffer';
+import BackgroundBlurOptions from './BackgroundBlurOptions';
+import BackgroundBlurProcessor from './BackgroundBlurProcessor';
+import { BlurStrengthMapper } from './BackgroundBlurStrength';
+import BackgroundBlurVideoFrameProcessorDelegate from './BackgroundBlurVideoFrameProcessorDelegate';
+import BackgroundBlurVideoFrameProcessorObserver from './BackgroundBlurVideoFrameProcessorObserver';
+import BackgroundFilterFrameCounter from './BackgroundFilterFrameCounter';
+import BackgroundFilterSpec from './BackgroundFilterSpec';
+
+/** @internal */
+class DeferredObservable<T> {
+  /** Access the last-resolved value of next() */
+  value: T | undefined = undefined;
+
+  private promise?: Promise<T>;
+  private resolve: (value: T) => void = null;
+
+  /** Create a promise that resolves once next() is called */
+  whenNext(): Promise<T> {
+    /* istanbul ignore else */
+    if (!this.promise) {
+      // externally-resolvable promise
+      this.promise = new Promise(resolve => (this.resolve = resolve));
+    }
+    return this.promise;
+  }
+
+  /** Update the value and resolve */
+  next(value: T): void {
+    // store the value, for sync access
+    this.value = value;
+    // resolve the promise so anyone awaiting whenNext resolves
+    this.resolve(value);
+    // delete the promise so future whenNext calls get a new promise
+    delete this.promise;
+  }
+}
+
+/**
+ * [[BackgroundBlurProcessorProvided]] implements [[VideoFrameProcessor]].
```

Comment:
BackgroundBlurProcessorProvided is implementing BackgroundBlurProcessor, hence, should we change this [[VideoFrameProcessor]] to [[BackgroundBlurProcessor]]?